metricbeat/helper/prometheus/textparse_test.go

```diff
@@ -144,6 +144,166 @@ process_cpu 20
 	require.ElementsMatch(t, expected, result)
 }
 
+func TestGroupWithHeaderBlockPrometheus(t *testing.T) {
+	input := `
+# HELP nginx_sts_server_bytes_total The request/response bytes
+# TYPE nginx_sts_server_bytes_total counter
+# HELP nginx_sts_server_connects_total The connects counter
+# TYPE nginx_sts_server_connects_total counter
+# HELP nginx_sts_server_session_seconds_total The session duration time
+# TYPE nginx_sts_server_session_seconds_total counter
+# HELP nginx_sts_server_session_seconds The average of session duration time in seconds
+# TYPE nginx_sts_server_session_seconds gauge
+# HELP nginx_sts_server_session_duration_seconds The histogram of session duration in seconds
+# TYPE nginx_sts_server_session_duration_seconds histogram
+nginx_sts_server_bytes_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",direction="in"} 0
+nginx_sts_server_bytes_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",direction="out"} 0
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="1xx"} 0
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="2xx"} 0
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="3xx"} 0
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="4xx"} 0
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="5xx"} 171
+nginx_sts_server_connects_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP",code="total"} 171
+nginx_sts_server_session_seconds_total{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP"} 0.016
+nginx_sts_server_session_seconds{listen="TCP:8091:127.0.0.1",port="8091",protocol="TCP"} 0.000
```

Comment:
[Suggestion]

2 things:

 - from [this](https://github.com/elastic/beats/pull/39743/files#diff-d66f15900f5676672ddf353458a9373cf8b1ab8e237c473191e4a64ae93cbba2R618-R621) the metric name of `nginx_sts_server_session_seconds_total` would be `nginx_sts_server_session_seconds` and the metric name of `nginx_sts_server_session_seconds` would be `nginx_sts_server_session`. Just checking, is my understand correct and is it the expected behaviour?
 - wouldn't be good to add 2 metrics with the same base name, like in your example `summary_metric_count` and `summary_metric_sum`?

Reply:
My example was for the metric family below. The metric family is called `summary_metric` and it includes various metrics such as `summary_metric_sum`, `summary_metric_count`, etc. We only want to store the metric type for the metric family `summary_metric` since all its metrics share the same type, which is `summary`.

```
# TYPE summary_metric summary
summary_metric{quantile="0.5"} 29735
summary_metric{quantile="0.9"} 47103
summary_metric{quantile="0.99"} 50681
summary_metric{noquantile="0.2"} 50681
summary_metric_sum 234892394
summary_metric_count 44000
summary_metric_impossible 123
# EOF
```

https://github.com/elastic/beats/blob/28ed783e6d6c90069ab1d7d5dd63839b0225ac17/metricbeat/helper/prometheus/textparse.go#L525-L535

`s` represents the metric family name (e.g., `summary_metric`). Only the metric family name is stored in the map, without its individual metrics.

For the examples you mentioned, `nginx_sts_server_session_seconds_total` and `nginx_sts_server_session_seconds`, these metrics are already the metric family names (they don't have any child metrics) and thus would not trigger the `!ok` condition. The `!ok` condition was added as an edge case for families that have multiple child metrics with different suffixes.